Company config&branch crud

oskprzym opened the pull request:
@Darthsmith zobacz proszę czy nie ma konfliktu w klasach BranchService itp. z Twoją częścią. Ja też zaraz sprawdzę.

Masakracja kodu oczywiście bardzo mile widziana :)

oskprzym committed:
Added BranchStatus (OPEN/CLOSED). Added a relevant field in Branch class.

This is to enable closing/opening branches from admin panel.

oskprzym committed:
Company config not developed yet.

Added CRUD for branch control - in terms of address etc. Simple frontend included. Easy navigation from /company/branches endpoint.
One bug known to me is present - trying to remove branch with id=1 (created on app startup), an exception is thrown (integrity constraints)

piotrkow95 commented on src/main/java/pl/sdacademy/carrental/controllers/CompanyController.java:
```diff
@@ -0,0 +1,67 @@
+import pl.sdacademy.carrental.services.CompanyService;
+
+@Controller
+@RequestMapping("/company") //Dostęp tylko dla admina! Dodać później
```
Walnąć "/admin/company" i po problemie. Będzie spójnie z inną adnotacją którą już zmieniłem wg. tej konwencji :)

MNowicki87 replied:
BTW - jak komentarze zaczynasz od `TODO`, to później łatwiej to znaleźć / śledzić w kodzie - w IntelliJ jest nawet zakładka na to

MNowicki87 replied:
👍

oskprzym replied:
Ok, tak zrobię, dzięki!

MNowicki87 commented on src/main/java/pl/sdacademy/carrental/domain/Company.java:
```diff
@@ -0,0 +1,71 @@
+    private String domain;
+
+
+    //Adres w Company - może dodać po prostu te pola, a nie jako odniesienie do Address? Bo OneToOne nie może być
+    //jednocześnie kiedy jest Address w Branch
+//    @NotNull
+//    @OneToOne
+//    @Column(name = "company_address")
+//    private Address address;
```
Dlaczego nie może być jednocześnie OneToOne?

oskprzym replied:
Bo kłóci się z OneToOne w Address w Branch... :/ Nie wiem jak to obejść, dlatego póki co zrobiłem tak. Zostawiłem to na konsultacje.

MNowicki87 commented on pom.xml:
```diff
@@ -11,7 +11,7 @@
     <artifactId>car-rental</artifactId>
     <version>0.0.1-SNAPSHOT</version>
-    <name>car-rental</name>
+    <name>car-company</name>
```
w sumie, nie mam nic przeciwko… a jest jakiś powód?

oskprzym replied:
Hmmmm, szczerze mówiąc, nie wiem dlaczego ta zmiana się wprowadziła. Ręcznie tego nie zmieniałem na pewno. Zostawiłbym tak, jak było od początku, czyli car-rental, dzięki za wychwycenie tego! Poprawiam na oryginalne :)

MNowicki87 commented on src/main/java/pl/sdacademy/carrental/domain/Company.java:
```diff
@@ -0,0 +1,71 @@
+public class Company {
+    private static Company INSTANCE;
+
+    @Id
+    private Long id = 1L;
```
Będzie tylko jedna - wg mnie id jest niepotrzebne

oskprzym replied:
Racja.

oskprzym replied:
Muszę dodać jednak jakieś Id (Hibernate tego wymaga) - proponuję name.

MNowicki87 commented on src/main/resources/templates/branches.html:
```diff
@@ -0,0 +1,32 @@
+<form method="get" th:action="@{/company/branches/create}">
+    <button type="submit">CREATE BRANCH</button>
+</form>
+<p>DO NOT remove branch with Id=1(created on app startup). For reasons unknown to me(Oskar), it throws and exception and I was unable to fix it. It will get fixed later.</p>
```
nie możesz usunąć brancha z powiązaniami (pracownicy/samochody/adres) - sql nie pozwala

oskprzym replied:
Aaaa czyli np. kaskadę trzeba by było. Ale to ciekawe, bo w delete dla innych branchów stworzonych już na późniejszym etapie (CRUDem) działa tak jak powinno.

oskprzym committed:
Hotfix:
-added /admin in CompanyController and relevant .html templates.
-fixed project name in pom.xml to car-rental.
-removed private Long id field in Company.class. Now name field is @Id.

MNowicki87 committed:
Update pom.xml

car-rental in <descpription>

MNowicki87 committed:
Merge branch 'master' into company-config&branch-CRUD
